wp/lib/bap_wp/src/precondition.mli

```diff
@@ -364,4 +368,21 @@ val set_sp_range : Env.t -> Constr.t
     all specified registers cannot point to the uninitalized stack region.
     They must be either below the bottom of the stack or above the initial
     stack pointer. *)
+
 val construct_pointer_constraint : Constr.z3_expr list -> Env.t -> (Constr.z3_expr list) option -> Env.t option -> Constr.t
+
+(** [loop_unroll num_unroll] returns a function that will unroll loops
+    [num_unroll] times. This is the default loop handler. *)
+val loop_unroll
+  : int ->
+  (Env.t -> Constr.t -> start:Bap.Std.Graphs.Ir.Node.t
+   -> Bap.Std.Graphs.Ir.t -> Env.t)
+
+(** [loop_invariant_checker invariants tid] returns a function that will check
+    the loop invariant for the loop that starts at [tid] given that the
+    invariant exists in [invariants]. *)
+val loop_invariant_checker
+  : string Bap.Std.Tid.Map.t
```

Comment:
Should we just create a type alias for this? Or name the argument?

Reply:
Something like `type loop_invariants = string Tid.Map.t`?

Reply:
Yep.